examples/jsm/controls/OrbitControls.js

```diff
@@ -558,10 +585,64 @@ var OrbitControls = function ( object, domElement ) {
 
 	}
 
+	var updateMouse3D = function () {
+
+		var v = new Vector3();
+		var v1 = new Vector3();
+
+		return function updateMouse3D( event ) {
+
+			var element = scope.domElement === document ? scope.domElement.body : scope.domElement;
+
+			if ( scope.object.isPerspectiveCamera ) {
+
+				v.set(
+				    ( event.clientX / element.clientWidth ) * 2 - 1,
+				    - ( event.clientY / element.clientHeight ) * 2 + 1,
+				    0.5 );
```

Comment:
@WestLangley Hi, I implemented this into my own orbitControl, but I found that it is not strictly zoom from cursor when u first start to zoom, since the mouse center point changes after zooming in or out. So i think using ` event.clientX / element.clientWidth` to calculate  could be improved.
![mod1](https://user-images.githubusercontent.com/33253363/106227525-4b80a980-6224-11eb-965e-bb0595a45a25.png)
![mod2](https://user-images.githubusercontent.com/33253363/106227529-4f143080-6224-11eb-8c1a-4712c5d3c2a9.png)

Reply:
hey bud could you be more speecific about what you did here i am trying to implement this also ! XD